src/main/java/org/openstreetmap/atlas/geography/boundary/CountryToShardListCache.java

```diff
@@ -0,0 +1,65 @@
+package org.openstreetmap.atlas.geography.boundary;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.geography.sharding.SlippyTile;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Country to shard list that is read from a .txt file. Useful for when using dense shard trees to
+ * reduce overhead of recalculating country/shard intersection.
+ *
+ * @author james-gage
+ */
+public final class CountryToShardListCache
+{
+    private static final Logger logger = LoggerFactory.getLogger(CountryToShardListCache.class);
+    private final HashMap<String, List<SlippyTile>> countryToShards = new HashMap<>();
+
+    public static CountryToShardListCache create(final Resource file)
+    {
+        return new CountryToShardListCache(file);
+    }
+
+    private CountryToShardListCache(final Resource file)
+    {
+        try
+        {
+            file.lines().forEach(line ->
+            {
+                final String[] countryAndShardList = line.split(Pattern.quote("||"));
```

Comment:
Nit: If you pull out the delimiter constant in the Cacher, than you reference it here, so if it ever changes, this code won't break.